### PR TITLE
Feature EPL-405: allow pre y post hooks when process orders

### PR DESCRIPTION
**Pull request by juabaez**

### CODE REVIEW

File: modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+            Order order = dao.getObject(Order.class, strC_Order_ID);
+            order.setDocumentAction(strdocaction);
+            OBDal.getInstance().save(order);
+            OBDal.getInstance().flush();
+
+            OBError msg = null;
+            for (ProcessOrderHook hook : hooks) {
+                msg = hook.preProcess(order, strdocaction);
+                if (msg != null && "Error".equals(msg.getType())) {
```
Comment (Matias-Bernal) [inline modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:67]: StringUtils.equals("Error",msg.getType())

Comment (Matias-Bernal) [reply modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:67]: Also, you may append all hook's messages in a single.

File: modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                    return msg;
+                }
+            }
+            // check BP currency
+            if ("CO".equals(strdocaction)) {
```
Comment (Matias-Bernal) [inline modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:72]: StringUtils.equals

File: modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    msg.setType("Error");
+                    msg.setTitle(Utility.messageBD(conn, "Error", vars.getLanguage()));
+                    msg.setMessage(String.format(errorMSG, order.getBusinessPartner().getId(),
+                            order.getBusinessPartner().getName()));
+
+                    return msg;
+                }
+            }
+
+            OBContext.setAdminMode(true);
```
Comment (Matias-Bernal) [inline modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:87]: It's must be in try statement

File: modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            } finally {
+                OBContext.restorePreviousMode();
+            }
+
+
+            final ProcessInstance pinstance = CallProcess.getInstance()
+                    .call(process, strC_Order_ID, null);
+
+
+            OBContext.setAdminMode();
```
Comment (Matias-Bernal) [inline modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:100]: It's must be in try statement

File: modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+                    return myMessage;
+                }
+            } finally {
+                OBContext.restorePreviousMode();
+            }
+
+            for (ProcessOrderHook hook : hooks) {
+                msg = hook.postProcess(order, strdocaction);
+                if (msg != null && "Error".equals(msg.getType())) {
```
Comment (Matias-Bernal) [inline modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:117]: StringUtils.equals("Error",msg.getType())

Comment (Matias-Bernal) [reply modules_core/org.openbravo.advpaymentmngt/src/org/openbravo/advpaymentmngt/ProcessOrderUtil.java:117]: Also, you may append all hook's messages in a single.

File: modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/ProcessOrders.java [hunk 6]
```diff
@@ -1,109 +1,116 @@
...
-        } catch (Exception e) {
-            OBDal.getInstance().rollbackAndClose();
-            return OBMessageUtils
-                    .translateError(DbUtility.getUnderlyingSQLException(e).getMessage());
+      log.debug("Process Invoice Action Parameters:");
+      log.debug(parameters.toString());
+
+      for (Order order : input) {
+        var message = processOrder(order, documentAction);
+        if (message.getType().equals("Error")) {
```
Comment (Matias-Bernal) [inline modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/ProcessOrders.java:67]: StringUtils.equals("Error",msg.getType())

File: modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/ProcessOrders.java [hunk 7]
```diff
@@ -1,109 +1,116 @@
...
+          errors++;
+        }
+        if (message.getMessage().isBlank()) {
```
Comment (Matias-Bernal) [inline modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/ProcessOrders.java:70]: StringUtils.isBlank

**Review by Matias-Bernal — CHANGES_REQUESTED** [review]
Fix, at least, setAdminMode errors

**Review by Matias-Bernal — APPROVED** [review]

**Review by Matias-Bernal — APPROVED** [review]

**Review by valeriag-etendo — APPROVED** [review]

**Pull request merged**